File: boj_1159.py
import sys
input = sys.stdin.readline

cnt = [0]*26
ans = []

n = int(input())
for i in range(n):
    word = input().strip()
    cnt[ord(word[0])-97] += 1

for i in range(26):
    if cnt[i] >= 5:
        ans.append(chr(i+97))

if not ans:
    print('PREDAJA')
else:
    print(*ans, sep='')

File: cpp_solutions/boj_1159/boj_1159.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main(){
    string drop_out = "PREDAJA";
    vector<int> cnt(26,0);
    bool b = false;
    int N;

    cin >> N;

    for(int i = 0; i < N; i++){
        string ss;
        cin >> ss;

        cnt[ss[0] - 'a']++;
    }

    for(int i=0;i<26;i++){
        if (cnt[i]>4) {
            cout << char('a' + i);
            b = true;
        }
    }

    if (!b) cout << drop_out;
}